app/controllers/journeys/adjustments/profitOrLoss/CurrentYearLossController.scala

```diff
@@ -44,51 +45,57 @@ class CurrentYearLossController @Inject() (override val messagesApi: MessagesApi
                                            formProvider: WhatDoYouWantToDoWithLossFormProvider,
                                            booleanFormProvider: BooleanFormProvider,
                                            whatDoYouWantToDoWithLossView: WhatDoYouWantToDoWithLossView,
-                                           carryLossForwardView: CarryLossForwardView)
+                                           carryLossForwardView: CarryLossForwardView)(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport {
 
   private val whatDoYouWantToDoWithLossPage = WhatDoYouWantToDoWithLossPage
   private val carryLossForwardPage          = CarryLossForwardPage
 
-  def onPageLoad(taxYear: TaxYear, businessId: BusinessId, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
+  def onPageLoad(taxYear: TaxYear, businessId: BusinessId, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
     implicit request =>
-      // TODO retrieve whether the user has other incomes from API 2085 and pension income in SASS-9566
-      val hasOtherIncomes = true
-      if (hasOtherIncomes) {
-        val filledForm = fillForm(whatDoYouWantToDoWithLossPage, businessId, formProvider(whatDoYouWantToDoWithLossPage, request.userType))
-        Ok(whatDoYouWantToDoWithLossView(filledForm, taxYear, businessId, request.userType, mode))
-      } else {
-        val filledForm = fillForm(carryLossForwardPage, businessId, booleanFormProvider(carryLossForwardPage, request.userType))
-        Ok(carryLossForwardView(filledForm, taxYear, businessId, request.userType, mode))
-      }
+      // TODO retrieve pension income in SASS-9566
+      val result = for {
+        hasOtherIncome <- service.hasOtherIncomeSources(taxYear, request.nino, request.mtditid)
```

Comment:
Since this is one line I'm not sure it needs to be in a for comprehension, it could just be a map

Reply:
Would be good to leave a placeholder for pension for clarity with the comment
```
val hasPensionIncome = true
val result = service.hasOtherIncomeSources(taxYear, request.nino, request.mtditid).map { hasOtherIncome =>
       if (hasOtherIncome && hasPensionIncome) {...
```

Reply:
same with onSubmit